### PR TITLE
Post run time plots to github artifacts

### PR DESCRIPTION
**Pull request by bparazin**
Test PR to see if everything works

### CODE REVIEW

**Comment by bparazin** [comment]
Wait, thought I was doing a PR to my branch. WIP for now

**Comment by bparazin** [comment]
Ok so with the cone search factoring out, there is currently nothing being benchmarked in the tests.

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        python-version: [3.6, 3.7, 3.8, 3.9]
+        python-version: [3.9 , 3.7, 3.8, 3.6]
```
Comment (lpsinger) [inline .github/workflows/build-and-test.yml:18]:
```suggestion
        python-version: [3.7, 3.8, 3.9]
```

**Review by lpsinger — CHANGES_REQUESTED** [review]

**Comment by lpsinger** [comment]
> Ok so with the cone search factoring out, there is currently nothing being benchmarked in the tests.

Yeah that's OK. There will be new tests very soon!

File: .github/workflows/runtime_plots.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import matplotlib.pyplot as plt
+import json
+import pandas as pd
```
Comment (lpsinger) [inline .github/workflows/runtime_plots.py:3]: Keep standard library imports and third party imports in separate blocks:

```suggestion
import json

import matplotlib.pyplot as plt
import pandas as pd
```

**Review by lpsinger — CHANGES_REQUESTED** [review]

File: .github/workflows/runtime_plots.py (hunk 2 above)
Comment (bparazin) [reply .github/workflows/runtime_plots.py:3]: :+1: Good to know!

File: .github/workflows/runtime_plots.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+with open('benchmark_results.json') as f:
+    data = json.load(f)
+
+df = pd.json_normalize(data['benchmarks'])
+df['func'] = df['name'].str.split('[', expand=True)[0]
+df['fullfunc'] = df['fullname'].str.split('[', expand=True)[0]
+df['param'] = pd.to_numeric(df['param'])
+for func, group in df.groupby('func'):
+    fig, ax = plt.subplots()
+    group.plot.scatter(x='param', y='stats.mean', yerr='stats.stddev', loglog=True, xlabel='N', ylabel='Runtime (s)', ax=ax)
```
Comment (lpsinger) [inline .github/workflows/runtime_plots.py:15]: Rewrap to <=79 chars per line

**Comment by bparazin** [comment]
The Ubuntu 3.9 check is going to fail since the JSON is empty atm; in the middle of adding error handling for it

**Comment by lpsinger** [comment]
> The Ubuntu 3.9 check is going to fail since the JSON is empty atm; in the middle of adding error handling for it

Don't worry about that. Like I said, there will be >0 tests shortly!

**Review by lpsinger — APPROVED** [review]

**Comment by lpsinger** [comment]
I don't think this is WIP anymore. Can you remove the WIP from the PR title?

**Comment by bparazin** [comment]
> > The Ubuntu 3.9 check is going to fail since the JSON is empty atm; in the middle of adding error handling for it
> 
> Don't worry about that. Like I said, there will be >0 tests shortly!

Oop, commited the error handling before I saw that; will revert

**Pull request merged**